base/abstractarray.jl

```diff
@@ -1554,12 +1554,16 @@ function (==)(A::AbstractArray, B::AbstractArray)
     if isa(A,AbstractRange) != isa(B,AbstractRange)
         return false
     end
+    anymissing = false
     for (a, b) in zip(A, B)
-        if !(a == b)
+        eq = (a == b)
+        if ismissing(eq)
+            anymissing = true
+        elseif !eq
             return false
         end
     end
-    return true
+    return anymissing ? missing : true
```

Comment:
could use `ifelse` here, similarly in `reduce.jl`

Reply:
That doesn't make any difference for simple cases like this:
```julia
julia> using Missings

julia> f(x) = x ? missing : true
f (generic function with 1 method)

julia> g(x) = ifelse(x, missing, true)
g (generic function with 1 method)

julia> @code_llvm f(true)

; Function f
; Location: REPL[1]:1
define nonnull %jl_value_t addrspace(10)* @julia_f_63991(i8) {
top:
  %1 = and i8 %0, 1
  %2 = icmp eq i8 %1, 0
  %. = select i1 %2, %jl_value_t addrspace(10)* addrspacecast (%jl_value_t* inttoptr (i64 139748733701392 to %jl_value_t*) to %jl_value_t addrspace(10)*), %jl_value_t addrspace(10)* addrspacecast (%jl_value_t* inttoptr (i64 139748736339184 to %jl_value_t*) to %jl_value_t addrspace(10)*)
  ret %jl_value_t addrspace(10)* %.
}

julia> @code_llvm g(true)

; Function g
; Location: REPL[2]:1
define nonnull %jl_value_t addrspace(10)* @julia_g_63993(i8) {
top:
  %1 = and i8 %0, 1
  %2 = icmp eq i8 %1, 0
  %3 = select i1 %2, %jl_value_t addrspace(10)* addrspacecast (%jl_value_t* inttoptr (i64 139748733701392 to %jl_value_t*) to %jl_value_t addrspace(10)*), %jl_value_t addrspace(10)* addrspacecast (%jl_value_t* inttoptr (i64 139748736339184 to %jl_value_t*) to %jl_value_t addrspace(10)*)
  ret %jl_value_t addrspace(10)* %3
}
```